addons/website/models/website_form.py

```diff
@@ -147,6 +149,28 @@ def init(self):
         self._cr.execute('ALTER TABLE ir_model_fields '
                          ' ALTER COLUMN website_form_blacklisted SET DEFAULT true')
 
+    @api.ondelete(at_uninstall=False)
+    def _check_if_used_in_website_form(self):
+        """prevent field deletion if used in a website form."""
+        arch_models = expression.OR([
+           ('arch_db', 'like', f' data-model_name="{model}"')
+        ] for model in self.mapped('model'))
+        form_views = self.env['ir.ui.view'].search(expression.OR([
+            [('type', '=', 'qweb')], arch_models
+        ]))
```

Comment:
If we really want to make this check, checking `ir.ui.view` is not enough, you have to check all HTML fields in the database, except some that would kill the perfs (e.g. mails). We have utils for that in the upgrade repo... but also apparently in standard with `_get_html_fields_attributes` (weird name :thinking:) (although it does seem to include the mail.message which it probably should not? :thinking:)